plugin/trino-mariadb/src/main/java/io/trino/plugin/mariadb/MariaDbClient.java

```diff
@@ -617,4 +652,236 @@ private static Optional<ColumnMapping> getUnsignedMapping(JdbcTypeHandle typeHan
+                    // low_value and high_value
+                    if (result.getMinValue() != null && result.getMaxValue() != null) {
+                        DoubleRange range = new DoubleRange(Double.parseDouble(result.getMinValue()), Double.parseDouble(result.getMaxValue()));
+                        columnStatisticsBuilder.setRange(range);
+                    }
+                }
+
+                tableStatistics.setColumnStatistics(column, columnStatisticsBuilder.build());
+            }
+            return tableStatistics.build();
+        }
+    }
+
+    private static class StatisticsDao
+    {
+        private final Handle handle;
+
+        public StatisticsDao(Handle handle)
+        {
+            this.handle = requireNonNull(handle, "handle is null");
+        }
+
+        Long getRowCount(JdbcTableHandle table)
+        {
+            return handle.createQuery("" +
+                            "SELECT cardinality FROM mysql.table_stats " +
+                            "WHERE db_name = :schema AND table_name = :table_name ")
+                    .bind("schema", table.getCatalogName())
+                    .bind("table_name", table.getTableName())
+                    .mapTo(Long.class)
+                    .findOne()
+                    .orElse(null);
+        }
+
+        List<ColumnStatisticsResult> getColumnStatistics(JdbcTableHandle table)
+        {
+            try {
+                handle.execute("SELECT 1 FROM mysql.column_stats WHERE 0=1");
+            }
+            catch (UnableToExecuteStatementException e) {
+                if (nullToEmpty(e.getMessage()).contains("Unknown table 'mysql.column_stats'")) {
+                    return new ArrayList<>();
+                }
+            }
+
+            return handle.createQuery("" +
+                            "SELECT column_name, avg_length, min_value, max_value, nulls_ratio, hist_type, " +
+                            "decode_histogram(hist_type, histogram) as histogram FROM mysql.column_stats " +
+                            "WHERE db_name = :schema AND table_name = :table_name")
+                    .bind("schema", table.getCatalogName())
+                    .bind("table_name", table.getTableName())
+                    .map((rs, ctx) -> new ColumnStatisticsResult(
+                            requireNonNull(rs.getString("column_name"), "column_name is null"),
+                            Optional.ofNullable(rs.getObject("avg_length", Float.class)),
+                            rs.getString("min_value"),
+                            rs.getString("max_value"),
+                            rs.getString("hist_type"),
+                            rs.getString("histogram"),
+                            Optional.ofNullable(rs.getObject("nulls_ratio", Float.class))))
+                    .list();
+        }
+    }
+
+    private static class ColumnStatisticsResult
+    {
+        private final String columnName;
```

Comment:
Remove this field - see other comment about `getColumnStatistics()`.

Reply:
Thanks for the review! The method _getColumnStatistics_ use the _getColumnName()_ method which need the `private final String columnName;`

Reply:
You can use `SimpleEntry` instead. Please refer to `MySqlClient.getColumnIndexStatistics`.

Reply:
Thanks for suggestion! changed refer to MySqlClient.getColumnIndexStatistics.